docs: add new section for CLI deprecations (#37332)
angular/angular: aio/content/guide/deprecations.md

- Clone the resulting value so that it has a new identity.
- Explicitly call [`ChangeDetectorRef.detectChanges()`](api/core/ChangeDetectorRef#detectchanges) to force the update. 

{@ decprecated-cli-flags}
{@ deprecated flags}

> ## Deprecated CLI flags

Reviewer: Maybe we can something like this section https://github.com/angular/angular/blob/00d02c416f6c4e277611e99d1a5e9b0f75405367/aio/content/guide/deprecations.md#index and provide more context like https://github.com/angular/angular/blob/00d02c416f6c4e277611e99d1a5e9b0f75405367/aio/content/guide/deprecations.md#angularcommon

Also maybe the title should be the followings, since since some of the mentioned are not flags.
```
Deprecated CLI APIs and options
```

Reviewer 2: I updated the title, but are you asking for individual sections for each package?

Also, in my opinion, saying "To help you future proof your apps..." isn't necessary. It reads like "filler" text. Could you elaborate on why more information might be needed here?

Reviewer: The filler text can be removed I was referring mainly to the overall structure of the sections and the splitting per package.

Reviewer 3: Thinking about this, what about adding subsections for each "functional area"?
For instance, there are currently build configuration options (`ng build`/`ng serve`/etc.), non-project tooling APIs, and schematic options (`ng generate`).  I think this may align well with where a user would find these deprecated elements.

Thoughts?

Reviewer 2: I've split the content into separate sections: one per package. If you'd like me to use different headings (to denote the functional areas, as @clydin suggests), let me know what headings I should use, as I don't know where all of these options live. Thanks!

- Clone the resulting value so that it has a new identity.
- Explicitly call [`ChangeDetectorRef.detectChanges()`](api/core/ChangeDetectorRef#detectchanges) to force the update. 

> {@ decprecated-cli-flags}

Reviewer 3: ```suggestion
{@ deprecated-cli-flags}
```

{@ deprecated flags}
## Deprecated CLI APIs and Options

This section contains a complete list all of the currently deprecated CLI flags.

| Package                                    | API/Option                      | May be removed in | Notes                                                                           |
| ------------------------------------------ | ------------------------------- | ----------------- |-------------------------------------------------------------------------------- |

> | @angular-devkit/build-angular:browser      | `i18nFile`                      | <!--v9--> v11     | Specified in the project locale configuration in version 9 and later.           |

Reviewer: The package is `@angular-devkit/build-angular`

| @angular-devkit/build-angular:karma        | `scripts[].lazy`                | <!--v8--> v11     | Renamed to `scripts[].inject`.                                                  |
| @angular-devkit/build-angular:karma        | `styles[].lazy`                 | <!--v8--> v11     | Renamed to `styles[].inject`.                                                   |
| @angular-devkit/build-angular:server       | `i18nFile`                      | <!--v9--> v11     | Specified in the project locale configuration in version 9 and later.           |
| @angular-devkit/build-angular:server       | `i18nFormat`        	           | <!--v9--> v11     | Format is now automatically detected.                                           |
| @angular-devkit/build-angular:server       | `i18nLocale`                    | <!--v9--> v11     | New [localization option](/guide/i18n#localize-config) in version 9 and later.  |
| @angular-devkit/build-angular:server       | `lazyModules`	                 | <!--v9--> v11     | Used with deprecated SystemJsNgModuleLoader.                                    |

> | @angular-devkit/build-angular:core         | `ModuleNotFoundException`       | <!--v8--> v10     | Redundant with native Node.js error code.                                       |

Reviewer: `@angular-devkit/build-angular:core` -> `@angular-devkit/:core`

> | @angular-devkit/:core                      | `ModuleNotFoundException`       | <!--v8--> v10     | Redundant with native Node.js error code.                                       |

Reviewer 3: ```suggestion
| @angular-devkit/core         | `ModuleNotFoundException`       | <!--v8--> v10     | Not used within projects - Tooling API only. Not Yarn PnP compatible and unused within the Angular CLI. Use Node.js [require.resolve](https://nodejs.org/api/modules.html#modules_require_resolve_request_options).                                       |
```
This should be used for `resolve`/`setResolveHook`/`ResolveOptions` as well.

Reviewer 2: Incorporated with minor changes for readability.

| @angular-devkit/build-angular:core         | `resolve`                       | <!--v8--> v10     | Redundant implementation of module resolution.                                  |
| @angular-devkit/build-angular:core         | `setResolveHook`                | <!--v8--> v10     | Redundant implementation of module resolution.                                  |
| @angular-devkit/build-angular:core         | `ResolveOptions`                | <!--v8--> v10     | Redundant implementation of module resolution.                                  |
| @angular-devkit/build-angular:core         | `terminal`                      | <!--v8--> v10     | Unused implementation of terminal codes (color).                                |

> | @angular-devkit/build-angular:core         | `isObservable`                  | <!--v8--> v10     | Redundant with official rxjs implementation.                                    |

Reviewer: rxjs -> RxJS

> | @angular-devkit/:core                      | `isObservable`                  | <!--v8--> v10     | Redundant with official RxJS implementation.                                    |

Reviewer 3: ```suggestion
| @angular-devkit/core         | `isObservable`                  | <!--v8--> v10     | Not used within projects - Tooling API only. Use `isObservable` function from the `rxjs` package.                                    |
```

Reviewer 2: Incorporated feedback with minor changes for readability.

| @ngtools/webpack                           | `discoverLazyRoutes`            | <!--v9--> TBD     | Used with deprecated SystemJsNgModuleLoader.                                    |
| @ngtools/webpack                           | `additionalLazyModules`         | <!--v9--> TBD     | Used with deprecated SystemJsNgModuleLoader.                                    |
| @ngtools/webpack                           | `additionalLazyModuleResources` | <!--v9--> TBD     | Used with deprecated SystemJsNgModuleLoader.                                    |

> | @schematics/angular:component              | `entryComponent`                | <!--v9--> TBD     | No longer needed with Ivy.                                                      |

Reviewer: Package name is @schematics/angular

- Clone the resulting value so that it has a new identity.
- Explicitly call [`ChangeDetectorRef.detectChanges()`](api/core/ChangeDetectorRef#detectchanges) to force the update. 

{@ deprecated-cli-flags}

> {@ deprecated flags}

Reviewer 4: ```suggestion
```

| `i18nFormat`                    | <!--v9--> v11     | Renamed to `format` to simplify the user experience.                            |
| `i18nLocale`                    | <!--v9--> v11     | Redundant with project’s source locale.                                         |
| `scripts[].lazy`                | <!--v8--> v11     | Renamed to `scripts[].inject`.                                                  |
| `styles[].lazy`                 | <!--v8--> v11     | Renamed to `styles[].inject`.                                                   |
| `i18nFile`                      | <!--v9--> v11     | Specified in the project locale configuration in version 9 and later.           |
| `i18nFormat`                    | <!--v9--> v11     | Format is now automatically detected.                                           |
| `i18nLocale`                    | <!--v9--> v11     | New [localization option](/guide/i18n#localize-config) in version 9 and later.  |
| `lazyModules`                   | <!--v9--> v11     | Used with deprecated SystemJsNgModuleLoader.                                    |

> ### @angular-devkit/:core

Reviewer 4: ```suggestion
### @angular-devkit/core
```
